platforms/documentation/docs/src/docs/userguide/reference/best-practices/best_practices_tasks.adoc

```diff
@@ -409,3 +409,60 @@ The output reveals that the `lib` project is now built when the `:app:goodClassp
 
 === Tags
 `<<tags_reference.adoc#tag:tasks,#tasks>>`, `<<tags_reference.adoc#tag:inputs-and-outputs,#inputs-and-outputs>>`, `<<tags_reference.adoc#tag:configurations,#configurations>>`
+
+[[dont_hardcode_task_names]]
+== Don’t Hardcode Task Names unless they are Documented as Public API
+
+Hardcoding task names can make a build fragile when upgrading Gradle or third-party plugins.
+
+=== Explanation
+
+Some task names in Gradle are part of the <<public_apis.adoc#public_gradle_apis,Gradle public API>> and are stable.
+Most other task names in Gradle and third-party plugins are internal implementation details. They may be renamed or removed in future versions and should not be relied upon.
+Task types are also not guaranteed to remain stable; plugin authors may refactor them. 
+
+In practice, configuring tasks by type is usually more robust than depending on specific names, especially in ecosystems like Android or Kotlin where many tasks are generated dynamically.
+
+Prefer, in order:
+
+1. *Plugin DSL* - Future-proof and most stable.
+2. *Task types* - Robust in many cases, but can change across plugin versions.
+3. *Task names* - Use only when they are explicitly documented as <<public_apis.adoc#public_gradle_apis,public API>>.
+
+As a plugin author, provide a DSL extension for users to configure behavior declaratively. Internally wire the configuration into your tasks. Avoid exposing task names as part of your public API unless you are prepared to maintain them with deprecation cycles.
+
+=== Example
+
+==== Don't Do This
+
+====
+include::sample[dir="snippets/bestPractices/avoidHardcodingTaskNames-avoid/kotlin",files="build.gradle.kts[tags=avoid-this]"]
+include::sample[dir="snippets/bestPractices/avoidHardcodingTaskNames-avoid/groovy",files="build.gradle[tags=avoid-this]"]
+====
+
+<1> Using `tasks.named` to get `GenerateMavenPom`: This relies on this task name and that's the only `GenerateMavenPom` task that needs configuring.
```

Comment:
🤔 This actually relies on name _and type_ explicitly in the Kotlin example (and implicitly in the Groovy one).

How about configuring something common to `Task` so that the type isn't necessary?

```
tasks.named("generatePomFileForMavenPublication").configure { 
    finalizedBy(someOtherTask)
}
```